File: Java/Java_Core/ListaExerciciosJavaPOO/Ex001/src/main/java/com/mycompany/ex1/Pessoa.java
package com.mycompany.ex1;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;

public class Pessoa {

    private String nome;
    private Date dataNascimento;
    private float altura;

    public String imprimeDados() {
        return "Nome: " + this.getNome() + " | Data de nascimento: "
                + this.getDataNascimento() + " | Altura: " + this.getAltura()
                + " m";
    }

    public String calculaIdade() {
        int anoAtual;
        int anoNasc;
        int mesAtual;
        int mesNasc;
        int idade;
        Calendar dataNasc = new GregorianCalendar();
        dataNasc.setTime(this.dataNascimento);
        mesNasc = dataNasc.get(Calendar.MONTH);
        anoNasc = dataNasc.get(Calendar.YEAR);

        //pegando o mês e ano atual
        Calendar hoje = Calendar.getInstance();
        mesAtual = hoje.get(Calendar.MONTH);
        anoAtual = hoje.get(Calendar.YEAR);

        //calculando a idade
        idade = anoAtual - anoNasc;
        if (mesAtual < mesNasc) {
            idade--;
        }
        return "A idade de " + this.nome + " é: " + idade + " anos.";
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDataNascimento() {
        SimpleDateFormat sf = new SimpleDateFormat("dd/MM/yyyy");
        String data = sf.format(this.dataNascimento);
        return data;
    }

    public void setDataNascimento(String nascimento) throws ParseException {
        SimpleDateFormat sf = new SimpleDateFormat("dd/MM/yyyy");
        this.dataNascimento = sf.parse(nascimento);
    }

    public float getAltura() {
        return altura;
    }

    public void setAltura(float altura) {
        this.altura = altura;
    }
}

File: Python/Alura/pythonOO/Ex01/pessoa.py
import datetime


class Pessoa:
    def __init__(self, nome, data_nascimento, altura):
        self.__nome = nome
        self.__data_nascimento = data_nascimento
        self.__altura = altura

    def imprimie_dados(self):
        print(
            "Nome: {} - Data de Nascimento: {} - Altura: {}".format(self.__nome, self.__data_nascimento, self.__altura))

    def calcula_idade(self):
        hoje = datetime.date.today()
        data_nascimento = datetime.datetime.strptime(self.__data_nascimento, "%d/%m/%Y")
        mes_atual = hoje.month
        ano_atual = hoje.year
        mes_nascimento = data_nascimento.month
        ano_nascimento = data_nascimento.year

        if mes_atual > mes_nascimento:
            idade = ano_atual - ano_nascimento
            print(f"{self.__nome} tem {idade} anos.")
        else:
            idade = int(ano_atual) - int(ano_nascimento) - 1
            print(f"{self.__nome} tem {idade} anos.")
